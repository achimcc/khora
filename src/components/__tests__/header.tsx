import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';

import Header from '../layout/header';
import GlobalStyle from '../layout/global';

const createNodeMock = (el: any) => {
	if (el.type === 'div') {
		const mockEl = document.createElement('div');
		return mockEl;
	}
	return null;
};

describe('Header', () => {
	it('matches snapshot', () => {
		const tree = renderer
			.create(
				<>
					<GlobalStyle key="one" />
					<Header key="two" siteSubTitle="Your favourite food delivery collective" />
				</>,
				{ createNodeMock }
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
	it('contains correct siteSubTitle', () => {
		render(<Header key="two" siteSubTitle="Your favourite food delivery collective" />);
		const text = screen.getByText('Your favourite food delivery collective');
		expect(text).toBeInTheDocument();
	});
});
