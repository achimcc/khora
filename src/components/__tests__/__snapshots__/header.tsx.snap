// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1em 0;
}

.c1 {
  margin-bottom: 0;
}

.c2 {
  -webkit-letter-spacing: 0.5em;
  -moz-letter-spacing: 0.5em;
  -ms-letter-spacing: 0.5em;
  letter-spacing: 0.5em;
  direction: rtl;
  text-indent: -0.5em;
  line-height: 1em;
}

.c3 {
  font-family: 'KhoraSymbols';
  display: inline;
}

.c4 {
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  word-spacing: 0.1em;
  font-size: 0.5em;
}

@media only screen and (max-width:500px) {
  .c1 {
    width: 85%;
  }
}

@media only screen and (min-width:500px) {
  .c1 {
    width: 80%;
  }
}

@media only screen and (min-width:600px) {
  .c1 {
    width: 60%;
  }
}

@media only screen and (min-width:1200px) {
  .c1 {
    width: 40%;
  }
}

<header
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="width: 100%;"
      >
        <div
          class="c2"
        >
          KH
          <div
            class="c3"
          >
            H
          </div>
          RA
        </div>
      </div>
      <div
        style="width: 100%;"
      >
        <div
          class="c4"
        >
          Your favourite food delivery collective
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
